### PR TITLE
docs: Config Approval Draft

### PR DESCRIPTION
**Pull request by ashokdevtron**
doc: Contains documentation of **Protect Configuration** feature (Config Approval)

### CODE REVIEW

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Also a suggestion, screenshots has some user personal details (emails & permissions), if we can blur or make it generic it would be better

File: docs/user-guide/creating-application/config-approval.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+In Devtron, these configurations are present in the **App Configuration** tab of your application.
+
+Any changes made to the following configurations will require approval:
+
+- Deployment Template
+- ConfigMaps
+- Secrets
+
+This stands true for both: base configuration and respective environment-level configuration.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/creating-application/config-approval.md:15]: Sure, will do. In this case, there's no glossary item for base configuration, we have base deployment template, Secret, and ConfigMap listed separately in glossary.

File: docs/user-guide/creating-application/config-approval.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+4. Once you have verified the changes, you can click **Deploy**.
+
+    ![Figure 10: Deploying App using New Config](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/config-approval/deploy.jpg)
+
+{% hint style="info" %}
+If you don't wish to deploy with the new changes, you can choose `Last deployed config` from the available drop-down.
+{% endhint %}
+
+## Enabling/Disabling Config Protection
```
Comment (abhibhaw) [inline docs/user-guide/creating-application/config-approval.md:97]: Who can enable and disable protection? RBAC missing

File: docs/user-guide/creating-application/config-approval.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+---
+
+## Approving the Configuration
+
+Only a different super-admin user or someone (who is not amongst the editors of the draft), having `Configuration approver` access, can approve the changes made to the configuration files as shown below.
+
+![Figure 6: Authorized User Approving the Changes](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/config-approval/approval-screen.jpg)
+
+If required, you can check whether a user has approval rights by going to **Global Configurations** → **Authorization** (dropdown) → **User Permissions**.
```
Comment (abhibhaw) [inline docs/user-guide/creating-application/config-approval.md:64]: Only superadmin can check permission

**Review by abhibhaw — DISMISSED** [review]

File: docs/user-guide/creating-application/config-approval.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/creating-application/config-approval.md:97]: Please help me with that information, @abhibhaw. I will add it to the doc.
I am not sure I know the answer...

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**